pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -431,6 +435,7 @@ public synchronized PinotResourceManagerResponse updateInstance(String instanceI
       return PinotResourceManagerResponse.failure("Instance " + instanceIdToUpdate + " does not exists");
     } else {
       InstanceConfig newConfig = InstanceUtils.toHelixInstanceConfig(newInstance);
+      newConfig.getRecord().setBooleanField(Helix.QUERIES_DISABLED, newInstance.getQueriesDisabled());
```

Comment:
Only set `Helix.QUERIES_DISABLED` when queries are disabled to be consistent with `addInstance()`